enclave_build/src/image.rs

```diff
@@ -69,3 +69,183 @@ pub fn deserialize_from_reader<R: Read, T: DeserializeOwned>(reader: R) -> Resul
 
     Ok(deserialized_obj)
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Unit tests should be in the same commit as the code they are testing, so that the user looking over a commit (this one for example) has full context on what's happening there.